utils/s3/credentials_providers/instance_profile_credentials_provider.cc

```diff
@@ -34,7 +34,7 @@ future<> instance_profile_credentials_provider::reload() {
 
 future<> instance_profile_credentials_provider::update_credentials() {
     auto factory = std::make_unique<utils::http::dns_connection_factory>(ec2_metadata_ip, port, false, ec2_md_logger);
-    http::experimental::client http_client(std::move(factory), 1, http::experimental::client::retry_requests::yes);
+    retryable_http_client http_client(std::move(factory), 1, {}, http::experimental::client::retry_requests::yes, retry_strategy);
```

Comment:
This is what I raised [here](https://github.com/scylladb/scylladb/pull/21933/commits/20caf7c7dd71d77126457727c6323782d6d4634a#r1963642123). The whole concept of this error handler is confusing in its existing implementation.

First, if the handler is empty, the exception is implicitly ignored.
Second, reading this line takes more effort to understand that `{}` here mean "ignore the exception".

Let's make the error_handler in the 3rd patch mandatory, and add an explicit `void ignore_exception()` helper that will be explicitly passed here instead of `{}` thing